src/main/java/edu/kit/kastel/sdq/intelligrade/state/PluginState.java

```diff
@@ -61,6 +63,35 @@ public static PluginState getInstance() {
         return pluginState;
     }
 
+    /**
+     * Logs out while displaying a warning if an assessment is still running
+     *
+     * @return true iff the logout was completed, false otherwise
+     */
+    public boolean logout() {
+        int answer = JOptionPane.OK_OPTION;
+        // check if confirmation is necessary because assessment is running
+        if (isAssessing()) {
+            answer = JOptionPane.showConfirmDialog(
```

Comment:
You should use IJ's dialogs, see https://github.com/kit-sdq/intelligrade/blob/de38e97ae095797964a4cf9231f455fbf4a5ec14/src/main/java/edu/kit/kastel/sdq/intelligrade/extensions/guis/ExercisePanel.java#L219-L221

Reply:
done